src/lua/protobuf.lua

```diff
@@ -657,9 +726,142 @@ end
 
 -- }}} Encoders
 
+-- {{{ Decoders
+
+local function read_unknown_field(ctx, type)
+    local res = string.sub(ctx.val, ctx.pos - 1, ctx.pos - 1)
+    local begin = ctx.pos
+    if type == WIRE_TYPE_LEN then
+        local len = wireformat.decode_int(ctx)
+        ctx.pos = ctx.pos + len
+    elseif type == WIRE_TYPE_I32 then
+        ctx.pos = ctx.pos + 4
+    elseif type == WIRE_TYPE_I64 then
+        ctx.pos = ctx.pos + 8
+    else
+        local _ = wireformat.decode_int(ctx)
+    end
+    res = res .. string.sub(ctx.val, begin, ctx.pos - 1)
+    return res
+end
+
+local function decode_repeated(protocol, field_def, ctx)
+    local res = {}
+    if ctx.packed then
+        local len = wireformat.decode_int(ctx)
+        local start_pos = ctx.pos
+        local value
+        while ctx.pos < start_pos + len do
+            ctx.ignore_repeated = true
+            value = decode_field(protocol, field_def, ctx)
+            table.insert(res, value)
+        end
+    else
+        repeat
+            local field_id
+        ctx.ignore_repeated = true
```

Comment:
First, the indent is broken here.

Next, I'm in doubt about passing the flag using the context this way. The idea is set the flag to go to a different branch in the next `decode_field` call and only in it. Neither keep it in processing of nested fields, nor in processing of next fields. However, the flag is unset in some not a very obvious place in another function.

I think that things would be much simpler if we just pass it to the function as a separate argument, like it is done in `encode_field`. This way it doesn't interfere with recursion anyhow and works just for the next call -- exactly what we need here.

Reply:
I saved previous implementation because I thought this movement will look strange, so I've restored it as it was before and fixed indentation.